Widescreen patch
Hoax - Cutscene: Elite
Relocated, combined, and modified Sandora soldier scripts
--- original
+++ modified
@@ -1,4 +1,6 @@
 entrypoint :ENTRYPOINT_0
+entrypoint :ENTRYPOINT_1
+entrypoint :ENTRYPOINT_2
 ENTRYPOINT_0:
 mov var[65], stor[23]
 incr stor[stor[stor[0], 23], 8]
@@ -12,3 +14,151 @@
 call 292, stor[0], 0x6000
 call 312, stor[0]
 rewind
+ENTRYPOINT_1:
+consume
+ENTRYPOINT_2:
+gosub_table stor[32], inl[:JUMP]
+consume
+#include ../../../../patches/libs/widescreen.txt
+JUMP:
+rel :JUMP_0
+rel :JUMP_1
+JUMP_0:
+call 101, stor[0], 0xe8, 0x16, 0xa8
+call 103, stor[0], 0x0, 0x400, 0x0
+call 96, 4
+mov 0, inl[:WS_WAIT_FLAG]
+mov 1, inl[:WS_WAIT_FLAG_MATCH]
+gosub inl[:WS_FLAG_WAIT]
+mov 13, inl[:WS_ANIMATION_0]
+mov 0xffffffff, inl[:WS_ANIMATION_1]
+gosub inl[:WS_ANIMATION]
+call 99, 0
+call 97, 13
+mov 13, inl[:WS_FRAMES]
+wait inl[:WS_FRAMES]
+call 99, 0
+call 97, 14
+mov 8, inl[:WS_FRAMES]
+wait inl[:WS_FRAMES]
+mov 0x122, inl[:WS_TARGET_X]
+mov 0x16, inl[:WS_TARGET_Y]
+mov 0xb4, inl[:WS_TARGET_Z]
+mov 6, inl[:WS_SPEED]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov 9, inl[:WS_FRAMES]
+wait inl[:WS_FRAMES]
+call 99, 1
+mov 12, inl[:WS_FRAMES]
+wait inl[:WS_FRAMES]
+call 2, 1, 1
+mov 0x144, inl[:WS_TARGET_X]
+mov 0x16, inl[:WS_TARGET_Y]
+mov 0xc4, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+gosub inl[:WS_MOVE]
+mov 0x1ba, inl[:WS_TARGET_X]
+mov 0xffffff4c, inl[:WS_TARGET_Y]
+mov 0x1d2, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov 0x229, inl[:WS_TARGET_X]
+mov 0xffffff4c, inl[:WS_TARGET_Y]
+mov 0x271, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov 0x320, inl[:WS_TARGET_X]
+mov 0xffffff4c, inl[:WS_TARGET_Y]
+mov 0x258, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+call 97, 0
+mov 0x14, inl[:WS_WAIT_FLAG]
+mov 1, inl[:WS_WAIT_FLAG_MATCH]
+gosub inl[:WS_FLAG_WAIT]
+mov 0x229, inl[:WS_TARGET_X]
+mov 0xffffff4c, inl[:WS_TARGET_Y]
+mov 0x271, inl[:WS_TARGET_Z]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+call 677, stor[0], inl[:WS_TARGET_X], inl[:WS_TARGET_Y], inl[:WS_TARGET_Z]
+gosub inl[:WS_MOVE]
+mov 0x1ba, inl[:WS_TARGET_X]
+mov 0xffffff4c, inl[:WS_TARGET_Y]
+mov 0x1d2, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov 0x144, inl[:WS_TARGET_X]
+mov 0x16, inl[:WS_TARGET_Y]
+mov 0xc4, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov 0x11a, inl[:WS_TARGET_X]
+mov 0x16, inl[:WS_TARGET_Y]
+mov 0xea, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 10, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+mov var[64][1], inl[:WS_WAIT_FLAG_SOBJ]
+call 102, inl[:WS_WAIT_FLAG_SOBJ], inl[:WS_TARGET_X], inl[:WS_TARGET_Y], inl[:WS_TARGET_Z]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+mov 10, inl[:WS_FRAMES]
+gosub inl[:WS_ROTATE]
+call 97, 11
+call 99, 1
+mov 0x16, inl[:WS_WAIT_FLAG]
+mov 1, inl[:WS_WAIT_FLAG_MATCH]
+gosub inl[:WS_FLAG_WAIT]
+mov 40, inl[:WS_FRAMES]
+wait inl[:WS_FRAMES]
+mov 0x81, inl[:WS_TARGET_X]
+mov 0x16, inl[:WS_TARGET_Y]
+mov 0xe8, inl[:WS_TARGET_Z]
+mov 8, inl[:WS_FRAMES]
+mov 0xffffffff, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+mov 8, inl[:WS_SPEED]
+mov 2, inl[:WS_ANIMATION_0]
+gosub inl[:WS_MOVE]
+call 102, var[64][1], inl[:WS_TARGET_X], inl[:WS_TARGET_Y], inl[:WS_TARGET_Z]
+mov 4, inl[:WS_FRAMES]
+mov 0, inl[:WS_ANIMATION_0]
+gosub inl[:WS_ROTATE]
+call 99, 0
+call 97, 11
+return
+JUMP_1:
+mov 0x1e, inl[:WS_WAIT_FLAG]
+mov 1, inl[:WS_WAIT_FLAG_MATCH]
+gosub inl[:WS_FLAG_WAIT]
+mov 14, inl[:WS_ANIMATION_0]
+mov 0xffffffff, inl[:WS_ANIMATION_1]
+gosub inl[:WS_ANIMATION]
+return
